neMag/Controllers/CategoriesController.cs

```diff
@@ -125,7 +128,7 @@ private IEnumerable<SelectListItem> GetAllCategories(Category me)
             });
             foreach (Category cat in (db.Categories).ToList())
             {
-                if (cat != me)
+                if (cat != me && cat.ParentId != me.CategoryId)
```

Comment:
Seems there is a problem here when "me" is set to null. It will generate an exception when accessing the CategoryId field.

Reply:
I tried to modify that line into `if (me is null || (cat != me && cat.ParentId != me.CategoryId))` and it seemed to work well afterwards.